python/ray/test/test_ray_init.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import pytest
+import redis
+
+import ray
+
+
+@pytest.fixture
+def start_ray_with_password():
+    ray.shutdown()
+
+    password = "some_password"
+    exception = None
+
+    # Fix for #3045
+    global f
+    f = ray.remote(f._function)
+
+    try:
+        info = ray.init(redis_password=password)
+    except Exception as e:
+        info = ray.init(redis_password=None)
+        exception = e
+    use_raylet = ray.global_state.use_raylet
+
+    return password, info, exception, use_raylet
+
+
+@pytest.fixture
+def use_credis():
```

Comment:
You can also just use 

https://github.com/ray-project/ray/blob/6240ccbc6edeafd2e3223be1edb1448f3ea5d0ca/test/runtest.py#L2165-L2167